packages/react-reconciler/src/ReactFiberCacheComponent.new.js

```diff
@@ -11,13 +11,34 @@ import type {ReactContext} from 'shared/ReactTypes';
 
 import {enableCache} from 'shared/ReactFeatureFlags';
 import {REACT_CONTEXT_TYPE} from 'shared/ReactSymbols';
-import AbortController from './ReactAbortController';
 
 import {pushProvider, popProvider} from './ReactFiberNewContext.new';
 import * as Scheduler from 'scheduler';
 
+// In environments without AbortController (e.g. tests)
+// replace it with a lightweight shim that only has the features we use.
+const AbortControllerLocal = enableCache
+  ? typeof AbortController !== 'undefined'
+    ? AbortController
+    : (function AbortControllerShim() {
+        this.listeners = [];
+        this.signal = {
+          aborted: false,
+        };
+
+        this.signal.addEventListener = (type, listener) => {
+          this.listeners.push(listener);
+        };
+
+        this.abort = () => {
+          this.signal.aborted = true;
+          this.listeners.forEach(listener => listener());
+        };
+      }: AbortController)
```

Comment:
This should minify better:

```js
    : (function AbortControllerShim() {
        const listeners = this.listeners = [];
        const signal = this.signal = {
          aborted: false,
        };

        signal.addEventListener = (type, listener) => {
          listeners.push(listener);
        };

        this.abort = () => {
          signal.aborted = true;
          listeners.forEach(listener => listener());
        };
      }: AbortController)
```

Reply:
Fixed, thanks for the tip - just curious how tf do you know this?

Reply:
Well, it's best to verify. But I'm just thinking, `this.listeners.push` can only stay `this.listeners.push`, but `listeners.push` can become `l.push` because the compiler can rename the variable.

Reply:
You can also skip the fake class and use closure methods:

```js
function createFallbackController() {
  const listeners = [];
  const signal = {
    aborted: false,
    addEventListener(type, listener) {
      listeners.push(listener);
    },
  };
  return {
    signal,
    abort() {
      signal.aborted = true;
      listeners.forEach(listener => listener());
    },
  };
}

const createController =
  typeof AbortController !== 'undefined'
    ? () => new AbortController()
    : createFallbackController;
```

Reply:
Done